R/gitstats_functions.R

```diff
@@ -326,21 +326,68 @@ get_users <- function(gitstats_object,
 #'     token = Sys.getenv("GITLAB_PAT_PUBLIC"),
 #'     orgs = "mbtests"
 #'   )
-#'  get_files(my_gitstats, c("LICENSE", "DESCRIPTION"))
+#'  get_files(
+#'    gitstats_obj = my_gitstats,
+#'    file_path = c("LICENSE", "DESCRIPTION"))
 #' }
 #' @return A data.frame.
 #' @export
-get_files <- function(gitstats_object,
-                      file_path,
-                      cache = TRUE,
-                      verbose = is_verbose(gitstats_object)){
-  gitstats_object$get_files(
+get_files_content <- function(gitstats_object,
+                              file_path,
+                              cache = TRUE,
+                              verbose = is_verbose(gitstats_object)) {
+  gitstats_object$get_files_content(
     file_path = file_path,
     cache = cache,
     verbose = verbose
   )
 }
 
+#' @title Get structure of files
+#' @name get_files_structure
+#' @description Pulls file structure for a given repository.
+#' @param gitstats_object A GitStats object.
+#' @param pattern An optional regular expression. If defined, it pulls file
+#'   structure for a repository matching this pattern.
+#' @param depth An optional integer. Defines level of directories to retrieve
+#'   files from. E.g. if set to `0`, it will pull files only from root, if `1`,
+#'   will take data from `root` directory and directories visible in `root`
+#'   directory. If left with no argument, will pull files from all directories.
+#' @param cache A logical, if set to `TRUE` GitStats will retrieve the last
+#'   result from its storage.
+#' @param verbose A logical, `TRUE` by default. If `FALSE` messages and printing
+#'   output is switched off.
+#' @examples
+#' \dontrun{
+#'  my_gitstats <- create_gitstats() %>%
+#'   set_github_host(
+#'     token = Sys.getenv("GITHUB_PAT"),
+#'     orgs = c("r-world-devs")
+#'   ) %>%
+#'   set_gitlab_host(
+#'     token = Sys.getenv("GITLAB_PAT_PUBLIC"),
+#'     orgs = "mbtests"
+#'   )
+#'  get_files_structure(
+#'    gitstats_obj = my_gitstats,
+#'    pattern = "\\.md"
+#'  )
+#' }
+#' @return A list of vectors.
+#' @export
+get_files_structure <- function(gitstats_object,
```

Comment:
Can we switch in results from API URL to Web URL? They're easier to understand for users, unless API is more convenient for further function calls e.g. in `getfiles_content()` 

![image](https://github.com/user-attachments/assets/8f4f3e7a-f086-4405-a6c7-888f184334ca)